corehq/apps/custom_data_fields/edit_entity.py

```diff
@@ -153,19 +154,32 @@ def make_fieldsets(self, form_fields, is_post, field_name_includes_prefix=False)
                     data_binds.append("select2: " + json.dumps([
                         {"id": id, "text": text} for id, text in field.widget.choices
                     ]))
-                field_names.append(Field(
-                    field_name,
-                    data_bind=", ".join(data_binds)
-                ))
+                if field_name.endswith(PROFILE_SLUG):
+                    profile_names.append(Field(
```

Comment:
If you go up one level (`init_form`), it looks like profile is handled separately and added in to the list of field names.  It'd probably be more straightforward to treat it separately at that level than to split it back out here.